src/tdastro/math_nodes/given_sampler.py

```diff
@@ -73,42 +77,108 @@ def compute(self, graph_state, rng_info=None, **kwargs):
         return results
 
 
-class TableSampler(FunctionNode):
-    """A FunctionNode that returns values from a table, including
-    a Pandas DataFrame or AstroPy Table.
+class GivenValueSampler(NumpyRandomFunc):
+    """A FunctionNode that returns randomly selected items from a given list
+    with replacement.
 
     Attributes
     ----------
+    values : list or numpy.ndarray
+        The values to select from.
+    _num_values : int
+        The number of values that can be sampled.
+    """
+
+    def __init__(self, values, seed=None, **kwargs):
+        self.values = np.asarray(values)
+        self._num_values = len(values)
+        if self._num_values == 0:
+            raise ValueError("No values provided for NumpySamplerNode")
+
+        super().__init__("uniform", seed=seed, **kwargs)
+
+    def compute(self, graph_state, rng_info=None, **kwargs):
+        """Return the given values.
+
+        Parameters
+        ----------
+        graph_state : `GraphState`
+            An object mapping graph parameters to their values. This object is modified
+            in place as it is sampled.
+        rng_info : numpy.random._generator.Generator, optional
+            A given numpy random number generator to use for this computation. If not
+            provided, the function uses the node's random number generator.
+        **kwargs : `dict`, optional
+            Additional function arguments.
+
+        Returns
+        -------
+        results : any
+            The result of the computation. This return value is provided so that testing
+            functions can easily access the results.
+        """
+        rng = rng_info if rng_info is not None else self._rng
+
+        if graph_state.num_samples == 1:
+            inds = rng.integers(0, self._num_values)
+        else:
+            inds = rng.integers(0, self._num_values, size=graph_state.num_samples)
+
+        return self.values[inds]
```

Comment:
I'm not sure if it is a better match here, but `rng.choice(values, size=...)` would do the same job as `self.value[rng.integers]`.

Reply:
Agreed. I'm not sure if one is better than the other, so I will leave it as-is for now. We can always change it in the future if needed.